pkg/globalmanager/incrementallearningjob.go

```
@@ -0,0 +1,794 @@
+			}
+		} else if podStatus == v1.PodSucceeded {
+			// watch pod status, if pod completed, set type completed
+			newConditionType = neptunev1.ILJobStageCondCompleted
+			klog.V(2).Infof("incrementallearning job %v/%v %v stage completed!", incrementaljob.Namespace, incrementaljob.Name, jobStage)
+		} else if podStatus == v1.PodFailed {
+			newConditionType = neptunev1.ILJobStageCondFailed
+			klog.V(2).Infof("incrementallearning job %v/%v %v stage failed!", incrementaljob.Namespace, incrementaljob.Name, jobStage)
+		}
+	case neptunev1.ILJobStageCondCompleted:
+		jobStage = getNextStage(jobStage)
+		newConditionType = neptunev1.ILJobStageCondWaiting
+
+	case neptunev1.ILJobStageCondFailed:
+		jobStage = neptunev1.ILJobTrain
+		newConditionType = neptunev1.ILJobStageCondWaiting
+
+	default:
+		// do nothing when given other type out of cases
+	}
+	klog.V(2).Infof("incrementallearning job %v/%v, conditions: %v", incrementaljob.Namespace, incrementaljob.Name, jobConditions)
+	if latestCondition.Type != newConditionType {
+		incrementaljob.Status.Conditions = append(incrementaljob.Status.Conditions, NewIncrementalJobCondition(newConditionType, jobStage))
+		needUpdated = true
+		return needUpdated, nil
+	}
+	return needUpdated, nil
+}
+
+// updateIncrementalJobStatus ensures that jobstatus can be updated rightly
+func (jc *IncrementalJobController) updateIncrementalJobStatus(incrementaljob *neptunev1.IncrementalLearningJob) error {
+	jobClient := jc.client.IncrementalLearningJobs(incrementaljob.Namespace)
+	var err error
+	for i := 0; i <= statusUpdateRetries; i = i + 1 {
+		var newIncrementalJob *neptunev1.IncrementalLearningJob
+		newIncrementalJob, err = jobClient.Get(context.TODO(), incrementaljob.Name, metav1.GetOptions{})
+		if err != nil {
+			break
+		}
+		newIncrementalJob.Status = incrementaljob.Status
+		if _, err = jobClient.UpdateStatus(context.TODO(), newIncrementalJob, metav1.UpdateOptions{}); err == nil {
+			break
+		}
+	}
+
+	// TODO
+	// return err
```

should return err instead of return nil

---

ok